src/cswinrt/code_writers.h

```diff
@@ -453,6 +453,71 @@ namespace cswinrt
             bind<write_parameter_name>(param));
     }
 
+    void write_projection_arg(writer& w, method_signature::param_t const& param)
+    {
+        w.write("%",
+            bind<write_parameter_name>(param));
+    }
+
+    void write_event_source_type_name(writer& w, type_semantics const& eventTypeSemantics)
+    {
+        auto eventTypeCode = w.write_temp("%", bind<write_type_name>(eventTypeSemantics, typedef_name_type::Projected, false));
+        
+        std::string eventTypeName = "_EventSource_" + eventTypeCode;
+        std::replace(eventTypeName.begin(), eventTypeName.end(), '<', '_');
```

Comment:
Would this be simpler to do with maybe a regex replace if possible or writing a custom write_type_name function?